app/dashboard/static/js/app/view-tests-case-id.2020.3.js

```diff
@@ -0,0 +1,383 @@
+        kernelNode.insertAdjacentHTML('beforeend', '&nbsp;');
+        kernelNode.appendChild(html.external());
+
+        if (!results.modules_dir || results.modules_dir == 'None') {
+            modulesNode = html.nonavail();
+        } else {
+            var modulesPath;
+
+            modulesPath = translatedURI[1] + "/" + results.modules_dir;
+            modulesNode = document.createElement('a');
+            modulesNode.appendChild(document.createTextNode(results.modules));
+            modulesNode.href =
+                translatedURI[0].path(modulesPath).normalizePath().href();
+            modulesNode.insertAdjacentHTML('beforeend', '&nbsp;');
+            modulesNode.appendChild(html.external());
+        }
+
+        html.replaceContent(
+            document.getElementById('defconfig'), defconfigNode);
+        html.replaceContent(
+            document.getElementById('endian'),
+            document.createTextNode(results.endian));
+        html.replaceContent(
+            document.getElementById('git-url'), gitNode);
+        html.replaceContent(
+            document.getElementById('compiler'),
+            document.createTextNode(results.compiler_version_full));
+        html.replaceContent(
+            document.getElementById('job-log'), logNode);
+        html.replaceContent(
+            document.getElementById('kernel-image'), kernelNode);
+        html.replaceContent(
+            document.getElementById('modules'), modulesNode);
+        html.replaceContent(
+            document.getElementById('dtb'), dtbNode);
+    }
+
+    function addRegressionHistory(results) {
+        var table;
+        var tbody;
+
+        table = document.getElementById('regression-table');
+        tbody = table.tBodies[0];
+
+        /* ToDo: highlight current revision row, and show "future" regressions
+         * when looking at other revisions (needs database schema changes).
+         */
+        results.regressions.reverse().forEach(function(data) {
+            var row = document.createElement('tr');
+            var createdOn = new Date(data.created_on.$date);
+            var cell;
+
+            cell = row.insertCell(-1);
+            cell.appendChild(document.createTextNode(data.kernel));
+            cell = row.insertCell(-1);
+            cell.appendChild(document.createTextNode(
+                createdOn.toDateString('%Y')
+            ));
+
+            cell = row.insertCell(-1);
+            cell.appendChild(document.createTextNode(
+                createdOn.getHours() + ':' +
+                createdOn.getMinutes() + ':' +
+                createdOn.getSeconds()
+            ));
+
+            cell = row.insertCell(-1);
+            cell.setAttribute('class', 'pull-center');
+            cell.appendChild(ttable.statusNode(data.status));
+
+            cell = row.insertCell(-1);
+            cell.setAttribute('class', 'pull-center');
+            cell.appendChild(ttable.detailsNode(
```

Comment:
What I wanted to do was to have the same table visible on the pages showing previous runs of the test, with the full regression history including "future" runs, and highlight the row matching the current view.  But this is proving a bit difficult with the current schema.  I guess we could still have something instead of the "details" link to show it is the current view, I'll think about it.

Reply:
Now the "details" links has been removed with the move to DataTables, each row has a link to a test case view.  I think the "current" row could have a different class to show with a different background or something like that.  I'll take a look if something simple can be done, but I don't think it's a blocker for this PR.

Reply:
This view is the one to show all the details for a single test case.